stablehlo/dialect/TypeInference.cpp

```diff
@@ -3072,8 +2992,7 @@ LogicalResult inferUniformQuantizeOp(
     SmallVectorImpl<ShapedTypeComponents>& inferredReturnShapes) {
   auto operandType = operand.getType().cast<ShapedType>();
```

Comment:
I think this is a good idea and I do plan to send a PR for that, but since this PR is already large I refrained from doing it. (Actually you may have noticed I did a few in the linalg legalization code, but that was because I needed to change the code anyway.)